src/main/kotlin/com/kotlindiscord/bot/extensions/CleanExtension.kt

```diff
@@ -0,0 +1,162 @@
+                        `in`.map { it.id.longValue }
+                    }
+
+                    val userIds = user?.map { it.id.longValue }
+                    val sinceTimestamp = since?.timestamp?.minusMillis(SINCE_OFFSET)
+
+                    logger.debug { cleanNotice }
+
+                    for (channelId in channels) {
+                        var query = bot.kord.cache.query<MessageData> {
+                            MessageData::channelId eq channelId
+
+                            if (!userIds.isNullOrEmpty()) run {
+                                MessageData::authorId `in` userIds
+                            }
+
+                            if (botonly) run {
+                                MessageData::authorIsBot eq true
+                            }
+
+                            if (!regex.isNullOrEmpty()) run {
+                                MessageData::content predicate { regex.all { regex -> regex.matches(it) } }
+                            }
+
+                            if (sinceTimestamp != null) run {
+                                MessageData::timestamp predicate {
+                                    Instant.parse(it).isAfter(sinceTimestamp)
+                                }
+                            }
+                        }.toCollection()
+
+                        if (count > 0) {
+                            query = query.sortedBy { Instant.parse(it.timestamp) }.reversed().slice(0..count)
+                        }
+
+                        if (query.size > MAX_DELETION_SIZE) {
+                            if (
+                                message.getAuthorAsMember()?.roles?.toList()
+                                    ?.contains(config.getRole(Roles.ADMIN)) == false
+                                // Basically, if the author isn't an admin.
+                            ) {
+                                message.channel.createMessage(
+                                    ":x: Cannot delete more than $MAX_DELETION_SIZE, " +
+                                            "please ask an admin to run this command with the `force:true` flag."
+                                )
+                                return@action
+                            } else {
+                                if (!force) {
+                                    message.channel.createMessage(
+                                    ":x: Cannot delete more than $MAX_DELETION_SIZE, " +
+                                            "run this command with the `force:true` flag to force it."
+                                    )
+                                    return@action
+                                }
+                            }
+                        }
+
+                        val cleanCount = "Messages to clean: ${query.joinToString(", ") { it.id.toString() }}"
+                        logger.debug { cleanCount }
+
+                        val channel = bot.kord.getChannel(Snowflake(channelId))
```

Comment:
If the variable is only used in the logger call, move it into the lambda. That's kind of the point of using a lambda.

Reply:
I forgot to leave the todo, I think we should have both logging replicated to the modlog.